Hardware/CMockHardwareMap.cpp

```diff
@@ -0,0 +1,297 @@
+        }
+        else
+        {
+            setrating(rating, channel);
+        }
+        b_command_recognised = true;
+    }
+    /**
+     * Command to modify heat capacity of the heater. Channel is optional and if
+     * omitted command is applied to all channels.
+     * >setcapacity(heater_capacity, radiator_capacity, [channel])
+     */
+    if (p_command->getName()->compare("setcapacity"))
+    {
+        uint8_t channel = (uint8_t)(*p_command)[2];
+        float heater_capacity = (*p_command)[0];
+        float radiator_capacity = (*p_command)[1];
+        if ((heater_capacity < 0) || (radiator_capacity < 0) ||
+            (channel > HARD_PWM_OUTPUTS))
+        {
+            p_comchannel->send("Arguments out of bounds.");
+        }
+        else
+        {
+            setcapacity(heater_capacity, radiator_capacity, channel);
+        }
+        b_command_recognised = true;
+    }
+    /**
+     * Command to modify heat conductance of the heater. Channel is optional and
+     * if omitted command is applied to all channels.
+     * >setconductance(heater_conductance, radiator_conductance, [channel])
+     */
+    if (p_command->getName()->compare("setconductance"))
+    {
+        uint8_t channel = (uint8_t)(*p_command)[2];
+        float heater_conductance = (*p_command)[0];
+        float radiator_conductance = (*p_command)[1];
+        if ((heater_conductance < 0) || (radiator_conductance < 0) ||
+            (channel > HARD_PWM_OUTPUTS))
+        {
+            p_comchannel->send("Arguments out of bounds.");
+        }
+        else
+        {
+            setconductance(heater_conductance, radiator_conductance, channel);
+        }
+        b_command_recognised = true;
+    }
+    return b_command_recognised;
+}
+
+void CMockHardwareMap::reset()
+{
+    mb_power_enable = false;
+    m_ambient_temperature = AMBIENT_T;
+    for (int i = 0; i < HARD_PWM_OUTPUTS; i++)
+    {
+        m_heater_power[i] = 0;
+        m_heater_rating[i] = HEATER_RATING;
+        m_temperature[i][0] = AMBIENT_T;
+        m_temperature[i][1] = AMBIENT_T;
+        m_heat_capacity[i][0] = HEATER_CAPACITY;
+        m_heat_capacity[i][1] = RADIATOR_CAPACITY;
+        m_heat_conductance[i][0] = HEATER_CONDUCTANCE;
+        m_heat_conductance[i][1] = RADIATOR_CONDUCTANCE;
+    }
+    m_incubator_temperature = m_ambient_temperature;
+    m_incubator_capacity = AMBIENT_T;
+    m_incubator_loss = INCUBATOR_LOSS;
+}
+
+void CMockHardwareMap::setrating(float rating, uint8_t channel)
+{
+    if (channel == 0)
+    {
+        for (int i = 0; i < HARD_PWM_OUTPUTS; i++)
+        {
+            m_heater_rating[i] = rating;
+        }
+    }
+    else
+    {
+        m_heater_rating[channel - 1] = rating;
+    }
+}
+
+void CMockHardwareMap::setcapacity(float heater_capacity,
+                                   float radiator_capacity,
+                                   uint8_t channel)
+{
+    if (channel == 0)
```

Comment:
Below you use `channel - 1`,  I guess it should be the same here: 
`if ( (channel - 1) == 0)`
Unless you are assuming that the parser will return 0 when trying to get an argument that doesn't exist (which I don't think is a good idea).

Reply:
zero means that optional parameter was omitted from the command. If optional parameter is omitted, then command is applied to all channels. This is correct logic.